modules/gapi/test/streaming/gapi_streaming_tests.cpp

```diff
@@ -0,0 +1,226 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+//
+// Copyright (C) 2019 Intel Corporation
+
+
+#include "../test_precomp.hpp"
+
+#include <opencv2/gapi/cpu/core.hpp>
+#include <opencv2/gapi/cpu/imgproc.hpp>
+
+#include <opencv2/gapi/fluid/core.hpp>
+#include <opencv2/gapi/fluid/imgproc.hpp>
+
+#include <opencv2/gapi/ocl/core.hpp>
+#include <opencv2/gapi/ocl/imgproc.hpp>
+
+namespace opencv_test
+{
+namespace
+{
+void initTestDataPath()
+{
+#ifndef WINRT
+    static bool initialized = false;
+    if (!initialized)
+    {
+        // Since G-API has no own test data (yet), it is taken from the common space
+        const char* testDataPath = getenv("OPENCV_TEST_DATA_PATH");
+        GAPI_Assert(testDataPath != nullptr);
+
+        cvtest::addDataSearchPath(testDataPath);
+        initialized = true;
+    }
+#endif // WINRT
+}
+
+cv::gapi::GKernelPackage OCV_KERNELS()
+{
+    static cv::gapi::GKernelPackage pkg =
+        cv::gapi::combine(cv::gapi::core::cpu::kernels(),
+                          cv::gapi::imgproc::cpu::kernels());
+    return pkg;
+}
+
+cv::gapi::GKernelPackage OCV_FLUID_KERNELS()
+{
+    static cv::gapi::GKernelPackage pkg =
+        cv::gapi::combine(OCV_KERNELS(),
+                          cv::gapi::core::fluid::kernels());
+    return pkg;
+}
+
+#if 0
```

Comment:
Need to document why OCL is disabled (some issue found with object lifetime or Mat/UMat relationship)

Reply:
Added